src/actions/VaccineActions.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2021
+ */
+
+import { ToastAndroid } from 'react-native';
+import { PermissionSelectors } from '../selectors/permission';
+import selectScannedSensors from '../selectors/vaccine';
+import { PermissionActions } from './PermissionActions';
+import BleService from '../bluetooth/BleService';
+import { syncStrings } from '../localization/index';
+import { UIDatabase } from '../database/index';
+
+export const VACCINE_ACTIONS = {
+  SCAN_START: 'Vaccine/sensorScanStart',
+  SCAN_STOP: 'Vaccine/sensorScanStop',
+  SENSOR_FOUND: 'Vaccine/sensorFound',
+};
+
+const scanStart = () => ({ type: VACCINE_ACTIONS.SCAN_START });
+const scanStop = () => ({ type: VACCINE_ACTIONS.SCAN_STOP });
+const sensorFound = macAddress => ({ type: VACCINE_ACTIONS.SENSOR_FOUND, payload: { macAddress } });
+
+const startSensorScan = () => async (dispatch, getState) => {
+  const bluetoothEnabled = PermissionSelectors.bluetooth(getState());
+  const locationPermission = PermissionSelectors.location(getState());
+  // Ensure the correct permissions before initiating a new sync process.
+  if (!bluetoothEnabled) await dispatch(PermissionActions.requestBluetooth());
+  if (!locationPermission) await dispatch(PermissionActions.requestLocation());
+
+  if (!bluetoothEnabled) {
+    ToastAndroid.show(syncStrings.please_enable_bluetooth, ToastAndroid.LONG);
```

Comment:
both of these strings include `We tried syncing temperatures, ` which isn't really relevant here.  Perhaps use `bluetooth_disabled` instead in this case.

Reply:
Oh thanks - forgot to double-check `syncStrings.json`! d5ab9069

Reply:
me too! 😆  only noticed because I broke something else while testing..